### DISCUSSION TITLE
How does the SepFormer handle arbitrary input lengths?

### DISCUSSION THREAD

**Discussion by hananbo26** [speechbrain/speechbrain#2226] [Q&A]
The SepFormer, introduced in the paper [ATTENTION IS ALL YOU NEED IN SPEECH SEPARATION](https://arxiv.org/pdf/2010.13154.pdf), is a popular speech separation network implemented in speechbrain.

However, having gone through the paper, I still need help understanding how it works. How does it deal with inputs of arbitrary lengths? Is there a maximal input length? And what happens if this maximal length is exceeded?

**Comment by mravanelli** [comment]
Hi,
you can take a look at the code and the training recipes available in
speechbrain here:
-
https://github.com/speechbrain/speechbrain/blob/develop/recipes/WSJ0Mix/separation/hparams/sepformer.yaml
-
https://github.com/speechbrain/speechbrain/blob/develop/speechbrain/lobes/models/dual_path.py
